### ISSUE TITLE
Typescript: Compile error when using getBufferAsync(Jimp.AUTO)

### ISSUE THREAD

**Issue by msdtim** [jimp-dev/jimp#878]
# Expected Behavior
`getBufferAsync(Jimp.AUTO)` should be allowed as per package README

> If Jimp.AUTO is passed as the MIME type then the original MIME type for the image (or "image/png") will be used. Alternatively, image.getMIME() will return the original MIME type of the image (or "image/png").

# Current Behavior
VSCode will report problem and tsc will throw compile error

# Failure Information (for bugs)
Compiler output:
> error TS2345: Argument of type '-1' is not assignable to parameter of type 'string'.

## Steps to Reproduce
npm install -g typescript
npm install --save jimp
npm install @types/node --save-dev
tsc -p tsconfig.json

## Context
In the `jimp.d.ts`, these functions should be updated to allow number type for `mime` parameter.
```
 getBase64(mime: string, cb: GenericCallback<string, any, this>): this;
 getBase64Async(mime: string): Promise<string>;
 getBuffer(mime: string, cb: GenericCallback<Buffer>): this;
 getBufferAsync(mime: string): Promise<Buffer>;
```

- Jimp Version: 0.10.2
- Operating System: macOS 10.14.6
- Node version: 12.16.1

**Comment by hipstersmoothie** [comment]
:rocket: Issue was released in [`v0.17.7`](https://github.com/jimp-dev/jimp/releases/tag/v0.17.7) :rocket:

**Comment by hipstersmoothie** [comment]
:rocket: Issue was released in [`v0.17.7`](https://github.com/jimp-dev/jimp/releases/tag/v0.17.7) :rocket:
